src/consortiaService.js

```diff
@@ -0,0 +1,153 @@
+  return localforage.getItem('okapiSess')
+    .then((sess) => {
+      sess.consortium = { ...(sess.consortium || {}), ...data };
+      return localforage.setItem('okapiSess', sess);
+    })
+    .then(({ consortium }) => {
+      store.dispatch(setConsortiumData(consortium));
+      if (consortium?.activeAffiliation) store.dispatch(setCurrentTenant(consortium.activeAffiliation.tenantId || defaultTenant));
+    });
+}
+
+/**
+ * fetchConsortia
+ *
+ * Retrieve a list of consortia.
+ *
+ * @param {string} okapiUrl
+ * @param {string} tenant
+ *
+ * @returns {Promise}
+ */
+export const fetchConsortia = (okapiUrl, tenant) => {
+  return fetch(`${okapiUrl}/consortia`, {
+    headers: {
+      'X-Okapi-Tenant': tenant,
+      'Content-Type': 'application/json',
+    },
+  });
+};
+
+/**
+ * fetchConsortiumTenants
+ *
+ * Retrieve a list of tenants related to the specified consortium.
+ *
+ * @param {string} okapiUrl
+ * @param {string} tenant
+ * @param {Object} options
+ * @param {string} options.consortiumId
+ *
+ * @returns {Promise}
+ */
+export const fetchConsortiumTenants = (okapiUrl, tenant, { consortiumId }) => {
+  return fetch(`${okapiUrl}/consortia/${consortiumId}/tenants`, {
+    headers: {
+      'X-Okapi-Tenant': tenant,
+      'Content-Type': 'application/json',
+    },
+  });
+};
+
+/**
+ * fetchConsortiumTenants
+ *
+ * Retrieve a list of affiliations between users and tenants in the specified consortium.
+ *
+ * @param {string} okapiUrl
+ * @param {string} tenant
+ * @param {string} token
+ * @param {Object} options
+ * @param {string} options.consortiumId
+ * @param {string} options.userId
+ *
+ * @returns {Promise}
+ */
+export const fetchConsortiumUserAffiliations = (okapiUrl, tenant, token, { consortiumId, userId }) => {
+  return fetch(`${okapiUrl}/consortia/${consortiumId}/user-tenants?userId=${userId}`, {
+    headers: {
+      'X-Okapi-Tenant': tenant,
+      'X-Okapi-Token': token,
+      'Content-Type': 'application/json',
+    },
+  });
+};
+
+/**
+ * fetchCurrentConsortiumData
+ *
+ * 1. Fetch data related to current consortium;
+ * 2. Update consortium in okapi session;
+ * 3. Update current tenant.
+ *
+ * @param {redux-store} store
+ * @param {object} data
+ *
+ * @returns {void}
+ */
+export const fetchCurrentConsortiumData = async (store, data) => {
+  try {
+    const centralTenant = okapi.tenant;
+    const { token, url } = store.getState().okapi;
+    const { consortia, totalRecords } = await fetchConsortia(url, centralTenant).then(resp => resp.json());
+    const consortium = totalRecords ? consortia[0] : {};
+    const consortiumId = consortium.id;
+
+    if (!consortiumId) return Promise.resolve();
+
+    const [
+      { tenants },
+      { userTenants: userAffiliations },
+    ] = await Promise.all([
+      fetchConsortiumTenants(
+        url,
+        centralTenant,
+        { consortiumId },
+      ).then(resp => resp.json()),
```

Comment:
@usavkov-epam it would be nice if `fetchConsortiumTenants` already did that for us so you can just get the results here without using `then`.